### PR TITLE
Add default templates and fix porting issues

### PR DESCRIPTION
**Pull request by schana**
https://phabricator.wikimedia.org/T164282#3256788

### CODE REVIEW

**Comment by schana** [comment]
@d00rman Would you mind reviewing this? Afterwords I'll send the requests to get the Gerrit repos setup.

File: config.dev.yaml [hunk 1]
```diff
@@ -75,6 +75,14 @@ services:
         query: '{{ default(request.query, {}) }}'
         headers: '{{request.headers}}'
         body: '{{request.body}}'
+      wdqsapi_req:
+        method: 'post'
+        uri: 'https://query.wikidata.org/bigdata/namespace/wdq/sparql'
+        headers:
```
Comment (d00rman) [inline config.dev.yaml]: Should also probably add `content-type: application/json` to the list of headers

File: lib/translation.js [hunk 2]
```diff
@@ -1,16 +1,104 @@
...
+        };
+    }
     app.conf.queries.article_tpl = new Template(app.conf.queries.article);
+
+    // set up the WDQS API request template
+    if (!app.conf.wdqsapi_req) {
+        app.conf.wdqsapi_req = {
+            method: 'post',
+            uri: 'https://query.wikidata.org/bigdata/namespace/wdq/sparql',
+            headers: {
```
Comment (d00rman) [inline lib/translation.js]: Idem, `content-type` should be here as well.

File: routes/translation.js [hunk 3]
```diff
@@ -16,15 +16,28 @@ const router = sUtil.router();
...
-function recommend(res, source, target, seed) {
-    return tUtil.recommend(app, source, target, seed)
-        .then((result) => {
-            result = result.slice(0, 24);
-            res.json({
-                count: result.length,
-                articles: result
+function recommend(req, res, source, target, seed) {
+    let count = 24;
+    if (Object.hasOwnProperty.call(req.query || {}, 'count')) {
```
Comment (d00rman) [inline routes/translation.js]: Easier: `if (req.query && req.query.count)`

File: routes/translation.js [hunk 4]
```diff
@@ -16,15 +16,28 @@ const router = sUtil.router();
...
+        count = parseInt(req.query.count, 10);
+        if (isNaN(count)) {
```
Comment (d00rman) [inline routes/translation.js]: You should also check that the number is between 1 and a reasonable max value

**Review by d00rman — CHANGES_REQUESTED** [review]
Looking good! A couple of minor issues and we should be good to go.

**Comment by schana** [comment]
Adding a content-type header to the WDQS requests results in a 405 response

**Comment by d00rman** [comment]
> Adding a content-type header to the WDQS requests results in a 405 response

This is weird. That should probably be reported to Discovery.

**Pull request merged**